articles/articles/article.php

```diff
@@ -1,176 +1,331 @@
 			}
 		}
-		
-		//get article data
-		/*if($article_id)
-		{
-			$artcl = new ContentModelArticle();
-			$adetails = $artcl->getItem($article_id);
-			//$art_obj->setState('filter.article_id',$article_id );
-		}*/
 
 		$rows = $art_obj->getItems();
 
-		//test code
-		/*JPluginHelper::importPlugin( 'content' );
-		$dispatcher = JEventDispatcher::getInstance();
-		$results = $dispatcher->trigger( 'onContentPrepare', array( 'com_content', &$row , ) );
-		*/
-		//
-
-		//format data
-		$obj = new BlogappSimpleSchema();
-		//$dispatcher = JEventDispatcher::getInstance();
-		//$item = $obj->mapPost($a_data,'', 100, array('text'));
-		foreach($rows as $row)
+		foreach ($rows as $subKey => $subArray)
 		{
-			if(!isset($row->text)||empty($row->text))
+			if ($subArray->catid)
 			{
-				$row->text = $row->introtext." ".$row->fulltext;
+				$subArray->catid = array('catid' => $subArray->catid,'title' => $subArray->category_title);
 			}
-			/*
-			$document	= JFactory::getDocument();
-			$document->setType('html');
-
-			JPluginHelper::importPlugin( 'content' );
-			$dispatcher = JEventDispatcher::getInstance();
-			$dispatcher->trigger('onContentPrepare', array ('com_content.article', &$row, &$row->params, null));
-			*/
-			
-			$item = $obj->mapPost($row,'', 100, array());
-		}
-	
-		$this->plugin->setResponse($item);
-		
+
+			if ($subArray->created_by)
+			{
+				$subArray->created_by = array('id' => $subArray->created_by,'name' => $subArray->author);
+			}
+
+			if ($subArray->images)
+			{
+				$images = json_decode($subArray->images);
+
+				foreach ($images as $key => $value)
+				{
+					if ($value)
+					{
+						$images->$key = JURI::base() . $value;
+					}
+				}
+
+				$subArray->images = $images;
+			}
+
+			unset($subArray->checked_out);
+			unset($subArray->checked_out_time);
+			unset($subArray->created_by_alias);
+			unset($subArray->modified_by);
+			unset($subArray->modified_by_name);
+			unset($subArray->urls);
+			unset($subArray->attribs);
+			unset($subArray->metadata);
+			unset($subArray->metakey);
+			unset($subArray->metadesc);
+			unset($subArray->xreference);
+			unset($subArray->readmore);
+			unset($subArray->category_route);
+			unset($subArray->category_access);
+			unset($subArray->category_alias);
+			unset($subArray->author_email);
+			unset($subArray->parent_title);
+			unset($subArray->parent_id);
+			unset($subArray->parent_route);
+			unset($subArray->parent_alias);
+			unset($subArray->rating);
+			unset($subArray->rating_count);
+			unset($subArray->published);
+			unset($subArray->parents_published);
+			unset($subArray->alternative_readmore);
+			unset($subArray->layout);
+			unset($subArray->params);
+			unset($subArray->displayDate);
+			unset($subArray->author);
+			unset($subArray->category_title);
+		}
+
+		$obj = new stdclass;
+
+		if (count($rows) > 0)
+		{
+			$obj->success = 'true';
+			$obj->{'data'} = array('results' => $rows, 'total' => count($rows));
```

Comment:
# data should be an object, not array
# total should be the total count that match the filters, not the total of items in the current set, i.e. if there are 240 articles matching the filters, and the API returns first 20 then the total should contain 240 not 20.